### PR TITLE
Fix resolve deps script

### LINKED ISSUE #3748: ext-auth-plugins Makefile doesn't work
**Describe the bug**
When trying to build an ext-auth-plugin using the provided `Makefile`, the `resolve-deps` target for determining correct versions for the go modules does not work.

**To Reproduce**
Follow steps described in https://github.com/solo-io/ext-auth-plugin-examples for building the plugin.

Summary below:
```
% echo $GLOOE_VERSION
1.5.1

% make resolve-deps
go run scripts/resolve_deps/main.go go.mod _glooe/dependencies
All shared dependencies match, writing new merged 'go.mod'
Writing go module file [go.mod], its content will replace your go.mod file

% make build
docker build --no-cache \
		--build-arg GO_BUILD_IMAGE=golang:1.15.2-alpine \
		--build-arg RUN_IMAGE=alpine:3.11 \
		--build-arg GLOOE_VERSION=1.5.1 \
		--build-arg STORAGE_HOSTNAME=storage.googleapis.com \
		--build-arg PLUGIN_MODULE_PATH=github.com/solo-io/ext-auth-plugin-examples \
		-t gloo-ext-auth-plugin-required_header-1.5.1:0.0.1 .
Sending build context to Docker daemon  61.88MB
Step 1/26 : ARG GO_BUILD_IMAGE

...

{"level":"error","ts":"2020-10-14T20:13:48.493Z","logger":"verify-plugins","caller":"scripts/verify_plugins.go:54","msg":"Plugin(s) cannot be loaded by Gloo","error":"failed to load plugin: failed to open plugin file: plugin.Open(\"plugins/RequiredHeader\"): plugin was built with a different version of package github.com/cncf/udpa/go/udpa/annotations","errorVerbose":"failed to load plugin\n\tplugins.(*loader).LoadAuthPlugin:/go/src/github.com/solo-io/solo-projects/projects/extauth/pkg/plugins/loader.go:52\nfailed to open plugin file\n\tmain.main:/go/src/github.com/solo-io/solo-projects/projects/extauth/scripts/verify_plugins.go:53\n\tmain.verifyPlugin:/go/src/github.com/solo-io/solo-projects/projects/extauth/scripts/verify_plugins.go:68\n\tmain.loadPlugin:/go/src/github.com/solo-io/solo-projects/projects/extauth/scripts/verify_plugins.go:91\n\tplugins.(*loader).LoadAuthPlugin:/go/src/github.com/solo-io/solo-projects/projects/extauth/pkg/plugins/loader.go:52\n\tplugins.(*loader).LoadAuthPlugin:/go/src/github.com/solo-io/solo-projects/projects/extauth/pkg/plugins/loader.go:50\n\tplugins.(*loader).loadPlugin:/go/src/github.com/solo-io/solo-projects/projects/extauth/pkg/plugins/loader.go:105\n\tplugins.glob..func1:/go/src/github.com/solo-io/solo-projects/projects/extauth/pkg/plugins/loader.go:24\nplugin.Open(\"plugins/RequiredHeader\"): plugin was built with a different version of package github.com/cncf/udpa/go/udpa/annotations","stacktrace":"main.main\n\t/go/src/github.com/solo-io/solo-projects/projects/extauth/scripts/verify_plugins.go:54\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:204"}
make: *** [Makefile:115: verify-plugin] Error 1
```

**Expected behavior**
`Makefile` works for determining correct `go.mod` file

**Additional context**
GlooE 1.5.1

### PR DESCRIPTION
**Pull request by kdorosh**
Also updates go.mod for Gloo Edge Enterprise 1.5.10.

The new resolve-deps is much simplified and was tested against Gloo Edge Enterprise versions 1.5.10, 1.4.16, and 1.3.14.
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/3748

### CODE REVIEW

File: changelog/v0.3.1/fix-resolve-deps.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+changelog:
+  - type: FIX
```
Comment (marcogschmidt) [inline changelog/v0.3.1/fix-resolve-deps.yaml]: I think this should be a breaking change, as we are removing exported functions/types from `pkg`. Someone might depend on them. Wdyt?

File: pkg/checks/compare.go [hunk 2]
```diff
@@ -1,10 +1,5 @@
 package checks
```
Comment (marcogschmidt) [inline pkg/checks/compare.go]: Is the `DependencyInfo` struct the only thing that is still used? If so I would just move it to `merge.go` and delete this file.

Comment (kdorosh) [reply pkg/checks/compare.go:1]: That would cause a circular dependency

Comment (marcogschmidt) [reply pkg/checks/compare.go:1]: Wait, why? They are in the same package

**Review by marcogschmidt — APPROVED** [review]

**Pull request merged**